pkg/kubectl/resource/visitor.go

```diff
@@ -396,19 +396,21 @@ func ExpandPathsToFileVisitors(mapper *Mapper, paths string, recursive bool, ext
 			if path != paths && !recursive {
 				return filepath.SkipDir
 			}
-			return nil
```

Comment:
This return eliminated the need for the new conditionals you added, right?

Actually, are any of your changes here needed?